### PR TITLE
Let users send and receive friend requests.

### PR DESCRIPTION
**Pull request by arringtonh**
Users can now send and accept friend requests by clicking on their name in a chat, then clicking a button in their profile. Users do not yet have a way to know who they are friends with, cannot deny friend requests, and cannot unfriend someone.

### CODE REVIEW

**Review by showermat — APPROVED** [review]

File: proj/src/main/webapp/chat.js [hunk 1]
```diff
@@ -501,11 +504,62 @@ function createMessageWithTemplate(key, messageObj) {
...
+        populateProfileSidebar(uid);
+    })
+}
 
+function friendRequestButton(uid) {
+    const currUserRef = firebase.database().ref('/users/'+currentUID+'/friend-requests/'+uid);
+    const otherUserRef = firebase.database().ref('/users/'+uid+'/friend-requests/'+currentUID);
+
+    const buttonHouse = document.getElementById('button-house');
+    const button = document.createElement('button');
```
Comment (aokeke) [inline proj/src/main/webapp/chat.js]: Why is this button being created here? Can you create it in the html and just display or hide it conditionally?

**Review by aokeke — CHANGES_REQUESTED** [review]

File: proj/src/main/webapp/chat.js (hunk 1 above)
Comment (arringtonh) [reply proj/src/main/webapp/chat.js]: fixed

File: proj/src/main/webapp/chat.js [hunk 2]
```diff
@@ -531,11 +534,59 @@ function createMessageWithTemplate(key, messageObj) {
...
+                // cancel friend request
+                button.innerText = 'cancel friend request';
+                button.onclick = function() {
+                    currUserRef.remove();
+                    otherUserRef.remove();
+                };
+                break;
+            case 'received':
+                // accept or deny
+                button.innerText = 'accept friend request';
```
Comment (aokeke) [inline proj/src/main/webapp/chat.js:568]: Where is the button for denying?

**Review by aokeke — CHANGES_REQUESTED** [review]

**Review by leeannachen — COMMENTED** [review]
lgtm

File: proj/src/main/webapp/chat.js (hunk 2 above)
Comment (arringtonh) [reply proj/src/main/webapp/chat.js:568]: yeah they can't deny yet, they can only accept or ignore

Comment (arringtonh) [reply proj/src/main/webapp/chat.js:568]: fixed

File: proj/src/main/webapp/chat.js [hunk 3]
```diff
@@ -531,11 +534,69 @@ function createMessageWithTemplate(key, messageObj) {
...
+                button.onclick = function() {
+                    currUserRef.remove();
+                    otherUserRef.remove();
+
+                    const currFriendRef = firebase.database().ref('/users/'+currentUID+'/friends/'+uid);
+                    const otherFriendRef = firebase.database().ref('/users/'+uid+'/friends/'+currentUID);
+                    currFriendRef.set(true);
+                    otherFriendRef.set(true);
+                };
```
Comment (aokeke) [inline proj/src/main/webapp/chat.js]: deny should also be hidden if they click on accept. Can you add a `deny.hidden=true;` to this block.

**Review by aokeke — CHANGES_REQUESTED** [review]

File: proj/src/main/webapp/chat.js (hunk 3 above)
Comment (arringtonh) [reply proj/src/main/webapp/chat.js]: fixed

**Review by aokeke — APPROVED** [review]
LGTM

**Pull request merged**